internal/controller/manifests_downloader.go

```diff
@@ -187,22 +201,81 @@ func (p *phaseReconciler) createManifestsConfigMap(ctx context.Context, metadata
 		configMap.SetAnnotations(map[string]string{compressedAnnotation: "true"})
 	}
 
-	gvk := p.provider.GetObjectKind().GroupVersionKind()
+	gvk := provider.GetObjectKind().GroupVersionKind()
 
 	configMap.SetOwnerReferences([]metav1.OwnerReference{
 		{
 			APIVersion: gvk.GroupVersion().String(),
 			Kind:       gvk.Kind,
-			Name:       p.provider.GetName(),
-			UID:        p.provider.GetUID(),
+			Name:       provider.GetName(),
+			UID:        provider.GetUID(),
 		},
 	})
 
-	if err := p.ctrlClient.Create(ctx, configMap); err != nil && !apierrors.IsAlreadyExists(err) {
-		return err
+	return configMap, nil
+}
+
+// OCIConfigMap templates config from the OCI source.
+func OCIConfigMap(ctx context.Context, provider operatorv1.GenericProvider, auth *auth.Credential) (*corev1.ConfigMap, error) {
+	store, err := FetchOCI(ctx, provider, auth)
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := store.GetMetadata(provider)
+	if err != nil {
+		return nil, err
+	}
+
+	components, err := store.GetComponents(provider)
+	if err != nil {
+		return nil, err
+	}
+
+	configMap, err := TemplateManifestsConfigMap(provider, ProviderLabels(provider), metadata, components, needToCompress(metadata, components))
+	if err != nil {
+		err = fmt.Errorf("failed to create config map for provider %q: %w", provider.GetName(), err)
+
+		return nil, err
+	}
+
+	if provider.GetUID() == "" {
+		// Unset owner references due to lack of existing provider owner object
+		configMap.OwnerReferences = nil
 	}
 
-	return nil
+	return configMap, nil
+}
+
+// RepositoryConfigMap templates ConfigMap resource from the provider repository.
+func RepositoryConfigMap(ctx context.Context, provider operatorv1.GenericProvider, repo repository.Repository) (*corev1.ConfigMap, error) {
+	metadata, err := repo.GetFile(ctx, provider.GetSpec().Version, "metadata.yaml")
+	if err != nil {
+		err = fmt.Errorf("failed to read metadata.yaml from the repository for provider %q: %w", provider.GetName(), err)
+
+		return nil, err
+	}
+
+	components, err := repo.GetFile(ctx, provider.GetSpec().Version, repo.ComponentsPath())
+	if err != nil {
+		err = fmt.Errorf("failed to read %q from the repository for provider %q: %w", repo.ComponentsPath(), provider.GetName(), err)
+
+		return nil, err
+	}
+
+	configMap, err := TemplateManifestsConfigMap(provider, ProviderLabels(provider), metadata, components, needToCompress(metadata, components))
+	if err != nil {
+		err = fmt.Errorf("failed to create config map for provider %q: %w", provider.GetName(), err)
+
+		return nil, err
+	}
+
+	if provider.GetUID() == "" {
+		// Unset owner references due to lack of existing provider owner object
```

Comment:
can you elaborate more why ownerRef needs to be set nil?

Reply:
It is due to providers being essentially a template, as there is no cluster kubeconfig used to create them. This leads to an owner reference without existing owner, causing the config map to be immediately deleted by garbage collection in the cluster once it gets applied